Profiler/Core/ProfilerPatch.cs

```diff
@@ -67,63 +60,89 @@ internal static class ProfilerPatch
             "DispatchSimulate",
         };
 
+#pragma warning restore 649
+        // ReSharper restore InconsistentNaming
+
+        #endregion
+
         private static readonly MethodInfo _generalizedUpdateTranspiler = ReflectionUtils.StaticMethod(typeof(ProfilerPatch), nameof(TranspilerForUpdate));
-        private static MethodInfo _distributedUpdaterIterate;
+
+        static readonly List<IProfiler> _observers = new List<IProfiler>();
+        public static ulong CurrentTick { get; private set; }
+
+        public const string GeneralEntrypoint = "General";
```

Comment:
Since all the entry points can be known at compile time I'd rather this was an enum.
Enum equality checks are also much, much faster.